### ISSUE TITLE
can't install gsl after GSL C installed

### ISSUE THREAD

**Issue by jianwu1** [RobinHankin/gsl#22]
Hi,

I'm trying to install the 'gsl' package but have been unsuccessful so far. 

Following the instruction, I have managed to install GSL C library and have tested that it's working:
<img width="314" alt="Screen Shot 2022-12-19 at 11 23 48 pm" src="https://user-images.githubusercontent.com/62498686/208438139-96da2cd7-e0bb-4417-baae-a8a67fffb439.png">
```
(base) [wu.j@rstudio-1 wu.j]$ gsl-config
Usage: gsl-config [OPTION]

Known values for OPTION are:

  --prefix              show GSL installation prefix 
  --libs                print library linking information, with cblas
  --libs-without-cblas  print library linking information, without cblas
  --cflags              print pre-processor and compiler flags
  --help                display this help and exit
  --version             output version information

An external CBLAS library can be specified using the GSL_CBLAS_LIB
environment variable. The GSL CBLAS library is used by default.

(base) [wu.j@rstudio-1 wu.j]$ gsl-config --prefix
/usr
```

However, I'm still unable to do install.packages("gsl"):
```
Installing package into ‘/stornext/Home/data/allstaff/w/wu.j/R/x86_64-pc-linux-gnu-library/4.2’
(as ‘lib’ is unspecified)
trying URL 'https://cran.rstudio.com/src/contrib/gsl_2.1-7.1.tar.gz'
Content type 'application/x-gzip' length 222357 bytes (217 KB)
==================================================
downloaded 217 KB

* installing *source* package ‘gsl’ ...
** package ‘gsl’ successfully unpacked and MD5 sums checked
** using staged installation
checking for gsl-config... /usr/bin/gsl-config
checking if GSL version >= 2.1... checking for gcc... /stornext/System/data/apps/gcc/gcc-11.1.0/bin/gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C... yes
checking whether /stornext/System/data/apps/gcc/gcc-11.1.0/bin/gcc accepts -g... yes
checking for /stornext/System/data/apps/gcc/gcc-11.1.0/bin/gcc option to enable C11 features... none needed
yes
configure: creating ./config.status
config.status: creating src/Makevars
** libs
/stornext/System/data/apps/gcc/gcc-11.1.0/bin/gcc -I"/stornext/System/data/apps/R/R-4.2.1/lib64/R/include" -DNDEBUG -I/usr/include -I.  -I/usr/local/include   -fpic  -g -O2  -c airy.c -o airy.o
/stornext/System/data/apps/gcc/gcc-11.1.0/bin/gcc -I"/stornext/System/data/apps/R/R-4.2.1/lib64/R/include" -DNDEBUG -I/usr/include -I.  -I/usr/local/include   -fpic  -g -O2  -c bessel.c -o bessel.o
/stornext/System/data/apps/gcc/gcc-11.1.0/bin/gcc -I"/stornext/System/data/apps/R/R-4.2.1/lib64/R/include" -DNDEBUG -I/usr/include -I.  -I/usr/local/include   -fpic  -g -O2  -c clausen.c -o clausen.o
/stornext/System/data/apps/gcc/gcc-11.1.0/bin/gcc -I"/stornext/System/data/apps/R/R-4.2.1/lib64/R/include" -DNDEBUG -I/usr/include -I.  -I/usr/local/include   -fpic  -g -O2  -c coulomb.c -o coulomb.o
/stornext/System/data/apps/gcc/gcc-11.1.0/bin/gcc -I"/stornext/System/data/apps/R/R-4.2.1/lib64/R/include" -DNDEBUG -I/usr/include -I.  -I/usr/local/include   -fpic  -g -O2  -c coupling.c -o coupling.o
/stornext/System/data/apps/gcc/gcc-11.1.0/bin/gcc -I"/stornext/System/data/apps/R/R-4.2.1/lib64/R/include" -DNDEBUG -I/usr/include -I.  -I/usr/local/include   -fpic  -g -O2  -c dawson.c -o dawson.o
/stornext/System/data/apps/gcc/gcc-11.1.0/bin/gcc -I"/stornext/System/data/apps/R/R-4.2.1/lib64/R/include" -DNDEBUG -I/usr/include -I.  -I/usr/local/include   -fpic  -g -O2  -c debye.c -o debye.o
/stornext/System/data/apps/gcc/gcc-11.1.0/bin/gcc -I"/stornext/System/data/apps/R/R-4.2.1/lib64/R/include" -DNDEBUG -I/usr/include -I.  -I/usr/local/include   -fpic  -g -O2  -c dilog.c -o dilog.o
/stornext/System/data/apps/gcc/gcc-11.1.0/bin/gcc -I"/stornext/System/data/apps/R/R-4.2.1/lib64/R/include" -DNDEBUG -I/usr/include -I.  -I/usr/local/include   -fpic  -g -O2  -c ellint.c -o ellint.o
ellint.c: In function ‘ellint_D_e’:
ellint.c:82:65: warning: passing argument 4 of ‘gsl_sf_ellint_D_e’ makes integer from pointer without a cast [-Wint-conversion]
   82 |     status[i] = gsl_sf_ellint_D_e(phi[i], k[i], sf_mode[*mode], &result) ;
      |                                                                 ^~~~~~~
      |                                                                 |
      |                                                                 gsl_sf_result * {aka struct gsl_sf_result_struct *}
In file included from ellint.c:1:
/usr/include/gsl/gsl_sf_ellint.h:84:66: note: expected ‘gsl_mode_t’ {aka ‘unsigned int’} but argument is of type ‘gsl_sf_result *’ {aka ‘struct gsl_sf_result_struct *’}
   84 | int gsl_sf_ellint_D_e(double phi, double k, double n, gsl_mode_t mode, gsl_sf_result * result);
      |                                                       ~~~~~~~~~~~^~~~
ellint.c:82:17: error: too few arguments to function ‘gsl_sf_ellint_D_e’
   82 |     status[i] = gsl_sf_ellint_D_e(phi[i], k[i], sf_mode[*mode], &result) ;
      |                 ^~~~~~~~~~~~~~~~~
In file included from ellint.c:1:
/usr/include/gsl/gsl_sf_ellint.h:84:5: note: declared here
   84 | int gsl_sf_ellint_D_e(double phi, double k, double n, gsl_mode_t mode, gsl_sf_result * result);
      |     ^~~~~~~~~~~~~~~~~
make: *** [ellint.o] Error 1
ERROR: compilation failed for package ‘gsl’
* removing ‘/stornext/Home/data/allstaff/w/wu.j/R/x86_64-pc-linux-gnu-library/4.2/gsl’
Warning in install.packages :
  installation of package ‘gsl’ had non-zero exit status

The downloaded source packages are in
	‘/tmp/RtmpZXcTFq/downloaded_packages’
```

I have specified in ~/.bashrc file and run `source ~/.bashrc`:
```
export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/stornext/HPCScratch/home/wu.j/gsl/lib
export C_INCLUDE_PATH=$C_INCLUDE_PATH:/stornext/HPCScratch/home/wu.j/gsl/include
```
But I still can't install "gsl" with `install.packages("gsl")`.

Can you give me some help please?

Best wishes,

Jian

**Comment by RobinHankin** [comment]
From the DESCRIPTION file

```
SystemRequirements: Gnu Scientific Library version >= 2.5
```

I ought to make this clearer in the package docs.

**Comment by jianwu1** [comment]
> From the DESCRIPTION file
> 
> ```
> SystemRequirements: Gnu Scientific Library version >= 2.5
> ```
> 
> I ought to make this clearer in the package docs.

Hi Robin,

Thanks for pointing this out to me!

The gsl-config PATH was still directed to the gsl version1.15. 
However, I have now changed the path to `gsl-config` to the local path where I installed gsl version2.7.1, and using this new version, gsl is still working properly:
<img width="878" alt="Screen Shot 2022-12-20 at 6 01 59 pm" src="https://user-images.githubusercontent.com/62498686/208607210-accacc77-3c61-4a87-ac4e-f6495381ca27.png">

However, when doing `install.packages("gsl")`, it seems the installation still seeks `gsl-config` in the old PATH `/usr/bin/gsl-config`, therefore causing failure of installation. 
<img width="954" alt="Screen Shot 2022-12-20 at 6 16 57 pm" src="https://user-images.githubusercontent.com/62498686/208607800-d3612ab7-54a4-4932-8513-d94a32ac5baf.png">
<img width="958" alt="Screen Shot 2022-12-20 at 6 17 11 pm" src="https://user-images.githubusercontent.com/62498686/208607832-83a6fb66-e9b8-4c07-8a37-b2f968a48452.png">

Can I ask how I can make `install.packages("gsl")` to use the new gsl-config which is the default in PATH now?
